pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/TableUpsertMetadataManager.java

```diff
@@ -41,7 +43,8 @@ public interface TableUpsertMetadataManager extends Closeable {
   void init(PinotConfiguration instanceUpsertConfig, TableConfig tableConfig, Schema schema,
       TableDataManager tableDataManager);
 
-  PartitionUpsertMetadataManager getOrCreatePartitionManager(int partitionId);
+  PartitionUpsertMetadataManager getOrCreatePartitionManager(int partitionId,
```

Comment:
Given `segmentOperationsThrottler` is instance level class, can we pass it in through `init()`?

Reply:
done